### PR TITLE
fix: react rules of hooks

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
- We abused the rules of hooks on this page a lot, which resulted in breaking it now.
- Decided to follow the right approach, confirmed it now works without breaking.

Screenshot 2023-12-12 at 10 04 19
Screenshot 2023-12-12 at 10 04 13

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-2040 #done

### CODE REVIEW

**Review by sshanzel — APPROVED** [review]
Definitely so much better now without all the comments.

Also, arguably we can store the handles in a single object since we only update it once for sanitization. This is optional though, not really part of the scope of this PR.

File: packages/webapp/components/layouts/ProfileLayout/index.tsx [hunk 1]
```diff
@@ -96,14 +91,18 @@ export default function ProfileLayout({
...
   const profile = fetchedProfile ?? initialProfile;
-  const isCurrentUserProfile = profile && profile.id === user?.id;
-
+  useEffect(() => {
+    if (profile) {
+      const purify = DOMPurify(window);
+      setTwitterHandle(sanitizeOrNull(purify, profile.twitter));
+      setGithubHandle(sanitizeOrNull(purify, profile.github));
+      setHashnodeHandle(sanitizeOrNull(purify, profile.hashnode));
+      setPortfolioLink(sanitizeOrNull(purify, profile.portfolio));
```
Comment (capJavert) [inline packages/webapp/components/layouts/ProfileLayout/index.tsx:102]: I don't think you need effect + state here, you can just `useMemo`.

**Review by capJavert — COMMENTED** [review]
Nice, just one optimization to avoid re-renders.

File: packages/webapp/components/layouts/ProfileLayout/index.tsx (hunk 1 above)
Comment (rebelchris) [reply packages/webapp/components/layouts/ProfileLayout/index.tsx:102]: You mean memo on the profile or the actual states?

Comment (rebelchris) [reply packages/webapp/components/layouts/ProfileLayout/index.tsx:102]: Hmm probably indeed, it's pretty much a edge case anyway not used that much.
I'll log it as a extra ticket if that's ok.

Comment (sshanzel) [reply packages/webapp/components/layouts/ProfileLayout/index.tsx:102]: Oh, right. Yep. One less re-render.

**Comment by omBratteng** [comment]
My face when something finally broke after we disregarded the warnings from the linter about hooks

**Pull request merged**